addons/web/controllers/main.py

```diff
@@ -1030,6 +1030,22 @@ def test_mobile_suite(self, mod=None, **kwargs):
     def benchmarks(self, mod=None, **kwargs):
         return request.render('web.benchmark_suite')
 
+    @http.route('/web/bundle/<string:bundle_name>', type='http', auth="user", methods=["GET"])
+    def bundle(self, bundle_name):
+        """
+        Request the definition of a bundle, including its javascript and css bundled assets
+        One might think that making this controller type=json is a good idea, but in fact odoo cannot receive Get
+        requests for routes with type=json that has no body, and the browser's fetch API cannot generate a Get request
+        that has a body.
+        """
+        all = request.env["ir.qweb"]._get_asset_nodes(bundle_name, js=True, css=True, async_load=True)
+        data = json.dumps([{
+            "type": tag,
+            "src": attrs["src"],
+        } for tag, attrs, _ in all])
+        return request.make_response(data,[('Content-Type','application/json'),
+                                     ('Cache-Control', 'max-age=%s' % http.STATIC_CACHE)])
```

Comment:
I don't think we want to cache this route, since there is no way to perform cache eviction on this route (no uniquifier). Since the returned asset urls contain uniquifiers, these will be loaded from cache when appropriate, so we will only have one call to this controller that could have been avoided, but almost no data will be transferred. In the long term my plan is to bake the lazy-loadable bundle-urls into the bundles that might trigger the lazy-loading, this way this controller can be removed entirely, but this requires some ground work and I don't think the trade-off of having no cache on this route in the mean time is horrible.

Reply:
you are right, we can't cache this, it would prevent asset invalidation.